### DISCUSSION TITLE
HPX users and applications

### DISCUSSION THREAD

**Discussion by msimberg** [STEllAR-GROUP/hpx#5368] [Show and tell]
This issue is meant to collect users of and/or applications using HPX. We will eventually place these on the HPX website (https://hpx.stellar-group.org). If you are developing a library or application and would like to share it with us and others, please comment below with a short summary of your project.

At least include the following:
- Who is developing the application (institution/company/personal project)

If you're willing to share the application as well include the following:
- Name of application
- What the application does
- Link to project repository or website

Feel free to add any other information that might be interesting as well.

**Comment by diehlpk** [comment]
@dmarce1 can you provide the information for Octo-Tiger?

**Comment by diehlpk** [comment]
@gentryx Can you provide the information for libgeocomp?

**Comment by diehlpk** [comment]
@diehlpk and @prashjha Can you provide the information for NLMech?

**Comment by weilewei** [comment]
Who is developing the application (institution/company/personal project):
**Oak Ridge National Laboratory, CSCS Swiss National Supercomputing Centre, Louisiana State University**
If you're willing to share the application as well include the following:

Name of application: **Dynamical Cluster Approximation (DCA++)**
What the application does: **A high-performance research software framework to solve quantum many-body problems with cutting edge quantum cluster algorithms**
Link to project repository or website: https://github.com/CompFUSE/DCA/

**Comment by venartas** [comment]
Who is developing the application (institution/company/personal project): **First Light Fusion Ltd**
Name of application: **Hytrac**
What the application does: **A Hydrodynamic Front-Tracking Code for the Study of High Energy Density Multi-Material Flows**
Link to project repository or website: **https://firstlightfusion.com/**

**Comment by hkaiser** [comment]
Who is developing the application (institution/company/personal project): **Louisiana State University**, **University of Oregon**, **University of Arizona**
Name of application: **Phylanx**
What the application does: **An Asynchronous Distributed C++ Array Processing Toolkit**
Link to project repository or website: http://phylanx.stellar-group.org/, https://github.com/STEllAR-GROUP/phylanx

**Comment by hkaiser** [comment]
Who is developing the application (institution/company/personal project): **Klaus Igelberger**
Name of application: **Blaze** 
What the application does: **A high performance C++ math library**
Link to project repository or website: https://bitbucket.org/blaze-lib/blaze/src/master/

**Comment by hkaiser** [comment]
Who is developing the application (institution/company/personal project): **University of Oregon**
Name of application: **APEX**
What the application does: **Autonomic Performance Environment for eXascale**
Link to project repository or website: https://github.com/khuck/xpress-apex

**Comment by hkaiser** [comment]
Who is developing the application (institution/company/personal project): **Los Alamos National Laboratory**
Name of application: **FleCSI**
What the application does: **The Flexible Computational Science Infrastructure**
Link to project repository or website: http://laristra.github.io/flecsi/

**Comment by kordejong** [comment]
Who is developing the application (institution/company/personal project): **Utrecht University, The Netherlands**, https://computationalgeography.org
Name of application: **LUE**
What the application does: **Environmental modelling framework**
Link to project repository or website: https://github.com/computationalgeography/lue

Our framework contains building blocks (data structures, algorithms) for developing scalable environmental models simulating biological and physical processes over time and through space. It is aimed at model developers, including domain experts, who may not know about the lower level details of software development, parallel computing and distributed computing.

**Comment by gentryx** [comment]
Who is developing the application: **STE||AR Group**
Name of application: **LibGeoDecomp**
What the application does: **LibGeoDecomp is an auto-parallelizing library to speed up your stencil code based computer simulations. It runs on virtually all current architectures, be it multi-cores, GPUs, or large scale MPI clusters.**
Link to project repository or website: libgeodecomp.org

**Comment by msimberg** [comment]
Who is developing the application: **Sandia National Labs and CSCS**
Name of application: **Kokkos**
What the application does: **Implements a programming model in C++ for writing performance portable applications targeting all major HPC platforms**
Link to project repository or website: https://github.com/kokkos/kokkos

Kokkos Core implements a programming model in C++ for writing performance portable applications targeting all major HPC platforms. For that purpose it provides abstractions for both parallel execution of code and data management. Kokkos is designed to target complex node architectures with N-level memory hierarchies and multiple types of execution resources. It currently can use CUDA, HPX, OpenMP and Pthreads as backend programming models with several other backends in development.

**Comment by diehlpk** [comment]
Who is developing the application?

LSU and UT Austin

Name of application:

NLMech

What the application does:

A simulation code for various non-local models, e.g. peridynamics.

Link to project repository or website

https://github.com/nonlocalmodels

We are a collaboration of researchers developing a simulation code for non-local models accelerated by the C++ standard library for parallelism and concurrency (HPX).

**Comment by diehlpk** [comment]
@msimberg Is this site available? I like to add it to the JOSS paper.

**Comment by msimberg** [comment]
@diehlpk this has not been put up yet. Given that we already have 10 entries in the comments here, I think we could already put a page with the applications. By when would you need it for the paper?

**Comment by msimberg** [comment]
@diehlpk in case this was not yet passed on to you: https://hpx.stellar-group.org/hpx-users/.

**Comment by arishah** [comment]
Who is developing the application: Cybersecurity Lab, Louisiana State University
Name of application: Gaslight
What the application does: grey-box fuzzer

Gaslight is a grey-box fuzzer, especially efficient for testing applications with large input files like memory forensics frameworks. Fuzz testing is performed using a custom file system to handle different sets of input mutations, and is guided by the results of tests as well as generated application outputs. The tests are distributed using HPX to run in parallel and scale the tests.

**Comment by kordejong** [comment]
The repository of the LUE project mentioned above moved to a new Github organization. It is now located at https://github.com/computationalgeography/lue instead of https://github.com/pcraster/lue.
